src/main/drivers/dshot.c

```diff
@@ -183,86 +186,125 @@ static uint32_t dshot_decode_eRPM_telemetry_value(uint16_t value)
 
 static void dshot_decode_telemetry_value(uint8_t motorIndex, uint32_t *pDecoded, dshotTelemetryType_t *pType)
 {
-    uint16_t value = dshotTelemetryState.motorState[motorIndex].rawValue;
-    const unsigned motorCount = motorDeviceCount();
+    uint16_t dshotDebugHighByte;
+    const uint16_t value = dshotTelemetryState.motorState[motorIndex].rawValue;
 
     if (dshotTelemetryState.motorState[motorIndex].telemetryTypes == DSHOT_NORMAL_TELEMETRY_MASK) {   /* Check DSHOT_TELEMETRY_TYPE_eRPM mask */
         // Decode eRPM telemetry
         *pDecoded = dshot_decode_eRPM_telemetry_value(value);
 
-        // Update debug buffer
-        if (motorIndex < motorCount && motorIndex < DEBUG16_VALUE_COUNT) {
+        // Update debug buffer (motorIndex < motorCount guaranteed by caller)
+        if (motorIndex < DEBUG16_VALUE_COUNT) {
```

Comment:
#define DEBUG16_VALUE_COUNT 8

:thinking: should we only update for motorCount (and make it global instead of a function call)

Reply:
That code is called from updateDshotTelemetry function, in a loop that that checks all existing motors.
The only place in the whole project where dshot_decode_telemetry_value is called is in this line:
![imagen](https://github.com/betaflight/betaflight/assets/2889851/6e7ce99e-c2d5-4a20-8514-d8fd811f073d)